xbmc/platform/linux/SysfsPath.cpp

```diff
@@ -19,19 +21,27 @@ bool CSysfsPath::Exists()
 }
 
 template<>
-std::string CSysfsPath::Get()
+std::optional<std::string> CSysfsPath::Get()
 {
-  std::ifstream file(m_path);
+  try
+  {
+    std::ifstream file(m_path);
 
-  std::string value;
+    std::string value;
 
-  std::getline(file, value);
+    std::getline(file, value);
 
-  if (file.bad())
+    if (file.bad())
```

Comment:
Is this if block even relevant anymore?

Reply:
I think it is, `std::ifstream::operator>>` doesn't throw if it can't convert the input if I understand correctly.